### PR TITLE
ng_udp: fix documentation

### PR DESCRIPTION
**Pull request by authmillenon**
Fixes and clarifies some documentation and also fixes the code to actually behave like documented.

### CODE REVIEW

File: sys/include/net/ng_udp.h [hunk 1]
```diff
@@ -86,7 +86,8 @@ int ng_udp_calc_csum(ng_pktsnip_t *hdr, ng_pktsnip_t *pseudo_hdr);
  * @param[in] dst_len       Length of @p dst, must be 2
  *
  * @return  pointer to the newly created (and allocated) header
- * @return  NULL on error
+ * @return  NULL on `src == NULL`, `dst == NULL`, `src_len != 2`, `dst_len != 2`
+ *          or on allocation error error
```
Comment (PeterKietzmann) [inline sys/include/net/ng_udp.h]: error error?

Comment (authmillenon) [inline sys/include/net/ng_udp.h]: Error error on the wall, who's the most critical of them all.

Fixed.

**Comment by PeterKietzmann** [comment]
ACK please squash

**Comment by authmillenon** [comment]
Squashed.

**Pull request merged**